### PR TITLE
fixes parenthesis

### LINKED ISSUE #639: Bug in filtering when using OR and grouping expressions with parentheses
Hi Meilisearch,

I found a bug when filtering using OR  while i am grouping expressions with parentheses. 

When i have a OR in my options query it looks to that OR and not any more to the other AND filters. 

See my screenshot with the query and the filters:

Search on: 'test'
Options: [filters] => "branche" = "32" AND ("gender" = "male" OR "gender" = "female")
)

Result found:
```
 Array
                (
                    [id] => 1
                    [firstname] => Test name
                    [middlename] => 
                    [lastname] => Lastname test
                    [gender] => female
                    [branche] => 39
)
```

Se the result above. Branche 39 shouldend be in the result. 

**Edit:** It results all rows that have gender = 'female' but it does not result rows that are 'male'. 

I also find out that it does not matter where my OR is. This bug is also shown when i put the OR in the begin of the query. 
 
**Edit:** I am using the PHP client.

### PR DESCRIPTION
**Pull request by MarinPostma**
fixes #639

### CODE REVIEW

**Comment by unknownhasher** [comment]
@MarinPostma thanks for fixing! Do you know then this is pulled to master? I need it for a release of today (a)

**Comment by erlend-sh** [comment]
We can't give a specific ETA. This is just one of many tasks the team will be working on today and the weeks to come.

If you need it for today you should just use a patched fork of your own making or the branch of this PR: https://github.com/MarinPostma/MeiliSearch/tree/fix_filter_parenthesis

**Comment by MarinPostma** [comment]
@unknownhasher I'm writing some more test before we merge it, thanks for finding this one out, that was a good catch

**Comment by unknownhasher** [comment]
I just tested this fix and it works. Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## v0.10.1 (unreleased)
 
+  - Add support for floatin points in filters (#640)
```
Comment (Kerollmops) [inline CHANGELOG.md:3]:
```suggestion
  - Add support for floating points in filters (#640)
```

**Pull request merged**